src/test/java/harness/utils/ExternalGCSBuckets.java

```diff
@@ -0,0 +1,110 @@
+package harness.utils;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.Identity;
+import com.google.cloud.Policy;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageClass;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.storage.StorageRoles;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+/** Utility methods for creating external GCS buckets for testing workspace references. */
+public class ExternalGCSBuckets {
+  public static final String gcpProjectId = "terra-cli-dev";
+
+  private static final String saKeyFile = "./rendered/ci-account.json";
+  private static final List<String> cloudPlatformScope =
+      Collections.unmodifiableList(Arrays.asList("https://www.googleapis.com/auth/cloud-platform"));
+
+  /**
+   * Get a bucket. This is helpful for testing controlled GCS bucket resources. It allows tests to
+   * check metadata that is not stored in WSM, only in GCS. This method takes in the credentials to
+   * use because tests typically want to check metadata as the test user.
+   */
+  public static Bucket getBucket(String bucketName, GoogleCredentials credentials)
+      throws IOException {
+    return getStorageClient(credentials).get(bucketName);
```

Comment:
I'm not seeing a lot of value in wrapping the `Storage` class. Why have each of these wrapping helper methods instead of using the `Storage` (or CRL's `StorageCow`) directly? `grantReadAccess` & `getStorageAccess` do some helping, but create/get/delete not as much.

Reply:
You can also use CRL to log the request/result. Currently it's at DEBUG, but you can configure the logger for the `StorageCow` class to INFO.